### ISSUE TITLE
Re: Content update question

### ISSUE THREAD

**Issue by hakansenturk** [uncatcrea/wp-appkit#264]
Hi, as you suggested I replaced the line App.navigateToDefaultRoute(); in functions.js with the following;

// Navigate to the default screen

     var currentScreenObject = App.getCurrentScreenObject();

// Navigate to the default screen (if we are not displaying a single) 
if ( currentScreenObject.screen_type !== 'single' ) {
    App.navigateToDefaultRoute();
}
it works :-), but the problem is after refresh is completed, I cannot go back to previous screen by pressing the back arrow button on the screen. But I can only go back by pressing the phone's back button. I could not find the solution for this.

**Our previous conversation about the issu as follows:**

Hi, after starting android application I created, the application begins to update wordpress post list. During this update process, If I enter a post from the list, it normally displays the post but after updating process is finishes, the program goes back to post list automatically.

In the previous version, I remember, I can read the post during the update and after the content update finishes.

I do not want the program goes back to post list after content update finishes. Is it any solution for this ?

**Your reply:**

Yes I believe so.

Displaying the default screen after refresh is done in the refresh:end event with the App.navigateToDefaultRoute(); in functions.js.

You can test if the current screen is a single and stop the routing to the default (refresh will still be done).

Here is a quick example on how to do it:

    var currentScreenObject = App.getCurrentScreenObject();

    // Navigate to the default screen (if we are not displaying a single)
    if ( currentScreenObject.screen_type !== 'single' ) {
        App.navigateToDefaultRoute();
    }
Of course, you can test for additional screen types if necessary.

**Comment by blupu** [comment]
Hi @hakansenturk well we've done some research and you clearly spotted a bug in WP-AppKit core. We have open an issue about it: https://github.com/uncatcrea/wp-appkit/issues/262.

In the meantime, you can use the _make-history_ filter in _functions.js_ to solve your problem.

```javascript
App.filter('make-history',function( history_action, history_stack, queried_screen, current_screen, previous_screen ){
	if( queried_screen.screen_type === 'single' && current_screen.screen_type === 'single' 
		&& queried_screen.item_id === current_screen.item_id ) { 
		//If we redisplay the same single (which happens after a refresh if we don't go back to default route), 
		//don't modify history:
		history_action = 'none';
	}
	return history_action;
});
```

**Comment by hakansenturk** [comment]
Hi @blupu thanks for the research , I'll try history filter to solve my problem. I'll write the results here, thanks a lot. I love wpappkit.

**Comment by hakansenturk** [comment]
ok. it works perfectly :-)

**Comment by mleroi** [comment]
Great! 
The fix for this issue (#262) will be merged in next WP-AppKit version.
When updating to this next version, the 'make-history' code that you used wont be needed anymore, but it will cause no harm if it stays in your theme.
